maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java

```diff
@@ -282,15 +287,42 @@ else if ( StringUtils.isNotEmpty( request.getMakeBehavior() ) )
     {
         List<MavenProject> result = projects;
 
-        if ( StringUtils.isNotEmpty( request.getResumeFrom() ) )
+        if ( StringUtils.isNotEmpty( request.getResumeFrom() ) || request.isResumeFromLastFailedProject() )
         {
             File reactorDirectory = null;
             if ( request.getBaseDirectory() != null )
             {
                 reactorDirectory = new File( request.getBaseDirectory() );
             }
 
-            String selector = request.getResumeFrom();
+            String selector;
+
+            if ( StringUtils.isNotEmpty( request.getResumeFrom() ) )
+            {
+                selector = request.getResumeFrom();
+            }
+            else
+            {
+                String buildDirectory = projects.stream()
```

Comment:
I would suggest to refactor this into a dedicated method, something like `determineWhereToResumeBuild()`.

Reply:
I have refactored this in a method now, but I called it `getLastFailedProject`. What do you think?

Reply:
Yes, I prefer the `getLastFailedProject`